py_zerox/zerox/processor/utils.py

```diff
@@ -41,3 +42,9 @@ def is_valid_url(string: str) -> bool:
         ]
     except ValueError:
         return False
+
+def sorted_nicely( l ): 
+    """ Sort the given iterable in the way that humans expect -> alphanumerically""" 
+    convert = lambda text: int(text) if text.isdigit() else text 
+    alphanum_key = lambda key: [ convert(c) for c in re.split('([0-9]+)', key) ] 
+    return sorted(l, key = alphanum_key)
```

Comment:
Does this work well with documents that have numeric characters in it? i.e.
```
final_document_version_032.pdf
```

It looks like it would, but I'd like to double check.

Reply:
@tylermaran, yes works very well. Worked well with whatever I threw at it it for years.

Reply:
The JS equivalent of this sorted_nicely function: 
```js
function sortedNicely(arr) {
    /**
     * Sort the given iterable in the way that humans expect -> alphanumerically.
     */
    const convert = (text) => {
        return /^\d+$/.test(text) ? parseInt(text) : text;
    };

    const alphanumKey = (key) => {
        return key.split(/([0-9]+)/).map(convert);
    };

    return arr.sort((a, b) => {
        const aKey = alphanumKey(a);
        const bKey = alphanumKey(b);

        for (let i = 0; i < Math.max(aKey.length, bKey.length); i++) {
            const aPart = aKey[i] || "";
            const bPart = bKey[i] || "";
            if (aPart !== bPart) {
                if (typeof aPart === "number" && typeof bPart === "number") {
                    return aPart - bPart;
                } else {
                    return aPart > bPart ? 1 : -1;
                }
            }
        }
        return 0;
    });
}

// Example usage:
const myList = ['file10', 'file2', 'file1', 'file20'];
console.log(sortedNicely(myList));
```
@tylermaran Let me know if this is something you would want to integrate in the JS SDK, i can raise a separate PR for that.